EIPS/eip-6780.md

```diff
@@ -27,37 +27,46 @@ The behaviour of `SELFDESTRUCT` is changed in the following way:
+   - Transfer the account balance to the target **and** set account balance to `0`.
+   - Note that no refund is given since [EIP-3529](./eip-3529.md).
+   - Note that the rules of [EIP-2929](./eip-2929.md) regarding `SELFDESTRUCT` remain unchanged.
+   - Note that when verkle tries are implemented on Ethereum, the cleared storage will be marked as having been written before but empty. This leads to no observable differences in EVM execution, but a contract having been created and deleted will lead to different state roots compared to the action not happening.
+
+A contract is considered created at the beginning of a create transaction or when a CREATE series operation begins execution (CREATE, CREATE2, and other operations that deploy contracts in the future).  If a balance exists at the contract's new address it is still considered to be a contract creation. 
 
 The `SELFDESTRUCT` opcode remains deprecated as specified in [EIP-6049](./eip-6049.md). Any use in newly deployed contracts is strongly discouraged even if this new behaviour is taken into account, and future changes to the EVM might further reduce the functionality of the opcode.
 
 ## Rationale
 
 Getting rid of the `SELFDESTRUCT` opcode has been considered in the past, and there are currently no strong reasons to use it. This EIP implements a behavior that will attempt to leave some common uses of `SELFDESTRUCT` working, while reducing the complexity of the change on EVM implementations that would come from contract versioning. A further option that was considered was to just remove storage clearing from `SELFDESTRUCT` in [EIP-6046](./eip-6046.md) while preserving removal of contracts, however this is not safe because existing contracts rely on storage being empty when they are deployed.
 
+Removing "hollow" accounts at addresses that are not externally owned and that only held a balance prior to contract creation provides a better user experience than leaving an account with a non-zero nonce, possible storage, and no code to access that storage.
+
 ## Backwards Compatibility
 
 This EIP requires a hard fork, since it modifies consensus rules.
 
-The only breaking change occurs when a contract is re-created at the same address using `CREATE2` (after a `SELFDESTRUCT`), where the `SELFDESTRUCT` is executed in a transaction that's different from the one that originally creates the contract.
+Contracts that depended on re-deploying contracts at the same address using `CREATE2` (after a `SELFDESTRUCT`) will no longer function properly if the created contract does not call `SELFDESTRUCT` within the current transaction.
+
+Previously it was possible to burn ether by calling `SELFDESTRUCT` targeting the executing contract as the beneficiary.  If the contract existed prior to the transaction the ether will not be burned.  If the contract was newly created in the transaction the ether will be burned, as before.
```

Comment:
Hi from Optimism.
FWIW this change does address our needs. 

The way we burn ETH is to deploy and fund [a contract that instantly SELFDESTRUCTs](https://github.com/ethereum-optimism/optimism/blob/a6e6988cbd89985a72c1fa2fcb9549e6a6ddfbba/packages/contracts-bedrock/contracts/libraries/Burn.sol#L28-L31).

Reply:
@maurelian, would it be hard to change the mechanism if we decide to disable the burn mechanism?

Reply:
AFAIK there is no other mechanism in the EVM for removing ETH from the state. Yes, Optimism could do something Optimism specific, but that would break our EVM equivalence.

IMO the ability to remove ETH from state is generally valuable, and has been [in use for a long time now](https://twitter.com/jeffehh/status/872871251366334464?s=20).

Reply:
I think the future will have no burning mechanism, but that depends on what OP advocates for.

Reply:
@shemnon 
if the core devs agree on disabling the burn mechanism in the future, then we should probably give a signal to the community in this EIP

Reply:
Deciding wether nor not to completely remove the burn I view as out of scope for the EIP and orthogonal to the issue at hand, so I don't think giving any indication one way or another in the text of the EIP is warranted.